src/trackinfoobject.cpp

```diff
@@ -722,20 +722,28 @@ QString TrackInfoObject::getURL() {
 }
 
 ConstWaveformPointer TrackInfoObject::getWaveform() {
+    QMutexLocker lock(&m_qMutex);
     return m_waveform;
 }
 
 void TrackInfoObject::setWaveform(ConstWaveformPointer pWaveform) {
+    QMutexLocker lock(&m_qMutex);
     m_waveform = pWaveform;
+    setDirty(true);
+    lock.unlock();
     emit(waveformUpdated());
```

Comment:
By grepping thru code, I found that `AnalyzerWaveform` is responsible for loading of an existing waveform in case the track is missing waveform.